checkers/rules/rules.go

```diff
@@ -736,3 +736,34 @@ func dynamicFmtString(m dsl.Matcher) {
 		Suggest("errors.New($f($*args))").
 		Report(`use errors.New($f($*args)) or fmt.Errorf("%s", $f($*args)) instead`)
 }
+
+//doc:summary Detects strings.Compare usage
+//doc:tags    performance experimental
```

Comment:
It's up to a debate whether this is a performance-first issue.
I would say most patterns would fall to style category, like using `==` instead of `strings.Compare`.
Other, less obvious patterns, can go to a separate, performance-tagged checker.

Reply:
changed to style